core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -520,4 +520,16 @@ package object config {
       .checkValue(v => v > 0, "The threshold should be positive.")
       .createWithDefault(10000000)
 
+  private[spark] val CREDENTIALS_RENEWAL_INTERVAL_RATIO =
+    ConfigBuilder("spark.security.credentials.renewalRatio")
+      .doc("Ratio of the credential's expiration time when Spark should fetch new credentials.")
+      .doubleConf
+      .createWithDefault(0.75d)
+
+  private[spark] val CREDENTIALS_RENEWAL_RETRY_WAIT =
+    ConfigBuilder("spark.security.credentials.retryWait")
+      .doc("How long to wait before retrying to fetch new credentials after a failure.")
+      .timeConf(TimeUnit.SECONDS)
+      .createWithDefaultString("1h")
```

Comment:
`.internal()` for both

Reply:
They're not really internal, just not documented.

Reply:
I thought that is what internal meant ... a user *could* specify them, but we don't document them at all, so not a stable part of the api etc.

Reply:
To me internal is something that is for internal Spark use, e.g. the configs I'm removing which are set by Spark itself.

Reply:
Is this "1h" too big if the token expire time is small, for example 8 hours, or even smaller, which will make the next retry directly fail.